src/Plugin.php

```diff
@@ -29,7 +29,7 @@ class Plugin {
    */
   public static function init() {
     add_filter('query_vars', __CLASS__ . '::query_vars');
-    add_filter('rewrite_rules_array', __CLASS__ . '::rewrite_rules_array', 100);
+    add_filter('product_rewrite_rules', __CLASS__ . '::product_rewrite_rules', 100);
```

Comment:
Can you add an inline comment to explain for posterity why you're intentionally using `product_rewrite_rules ` instead of `rewrite_rules_array`?  Also, is the priority still correct with this change; do you still want to append?